### ISSUE TITLE
Complete debugger Output buffer should be copyable so it can be pasted elsewhere

### ISSUE THREAD

**Issue by wyattbiker** [godotengine/godot-proposals#133]
**Describe the problem or limitation you are having in your project:**
I am using Mac OS X.
Im dumping out large arrays, into the output window using print().  However I would like to copy the complete output from that window to a text editor to review it.

It does not allow me to highlight and copy/paste the complete Output.  ctrl+a or command+a do not work.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
Will allow large output to be copied/pasted to another text editor or file.

**Comment by Calinou** [comment]
There's a **Copy** button at the top-right of the Output panel:

![image](https://user-images.githubusercontent.com/180032/66277721-df499c00-e8a2-11e9-9dc3-12bce2c5b197.png)

I'm not sure if it's in 3.1.1 though, it may only have been added to the `master` branch (which means it'll be available in 3.2).

**Comment by wyattbiker** [comment]
My bad I did not mention version I am using 3.1.1.stable.official.

No Copy button. But I'm glad they added that in for 3.2.

**Comment by hammeron-art** [comment]
The output is a RichTextLabel. RichTextLabel supports selection but nothing can be done with it. You can't even get the selected text.

Instead of an editor only fix, RichTextLabel could implement a few features available in TextEdit such as `Shortcut Keys Enable` and its selection all, copy and paste.

**Comment by Calinou** [comment]
@hammeron-art RichTextLabel is read-only, which means it can't support a Paste operation :wink:

**Comment by hammeron-art** [comment]
@Calinou Yeah, my bad. I mean read-only operations like selection all and copy.

**Comment by golddotasksquestions** [comment]
> However I would like to copy the complete output from that window to a text editor to review it.

Meanwhile, instead of using print, you could write your output to a text file.

**Comment by dalexeev** [comment]
I think this proposal can be considered resolved. The "Copy Selection" button in the sidebar has long had a hidden behavior of copying all text if nothing is selected. Once upon a time, you couldn't even select text in the Output Log. Now there is a context menu with the appropriate items. Finally, the behavior of the "Copy" item will probably be adjusted so that you don't even need to select all text.

![](https://github.com/user-attachments/assets/9f5d2c03-0663-4623-998a-129e2cce6eb4)
![](https://github.com/user-attachments/assets/8d0aed9b-3caf-4c4b-a432-26386852f047)

* https://github.com/godotengine/godot/pull/42066
* https://github.com/godotengine/godot/pull/74114
* https://github.com/godotengine/godot/pull/106404
